roles/backup/defaults/main.yml

```diff
@@ -16,3 +16,7 @@ pulp_admin_password_secret: "{{ pulp_name }}-admin-password"
 postgres_configuration_secret: "{{ pulp_name }}-postgres-configuration"
 
 custom_resource_key: '_pulp_pulpproject_org_pulpbackup'
+
+database_type: 'unmanaged'
```

Comment:
I didn't get this, could you please explain?

Reply:
The user has the option of supplying the postgres secret to be used if they already have a database (RDS, crunchydata, etc)
https://github.com/pulp/pulp-operator/blob/main/deploy/crds/pulpproject_v1beta1_pulp_crd.yaml#L59

However, if the operator generates the postgres secret it is marked with a type `managed`:
https://github.com/pulp/pulp-operator/blob/main/roles/postgres/templates/postgres.secret.yaml.j2#L14

If the user supplies their own secret it is considered `unmanaged`. The type isn't required for pulp to deploy correctly, so if the user doesn't supply this value in their secret it should not break.

The backup code defaults to `unmanaged`, however, if the `type` value is present in the given secret it will be overridden: https://github.com/pulp/pulp-operator/pull/111/files#diff-a29bce4ae2beeb7039f9e579e788bf0cdcf975514ce5e7674f5e2b3ba0984e58R71

In the current template for backup it expects this value to be present so it needed a default:
https://github.com/pulp/pulp-operator/pull/111/files#diff-f068ab4b2941a8726510f82562ec212dc435ace02223c9c24e215d28d7857119R16

Reply:
thank you for the explanation!